### DISCUSSION TITLE
Copilot for students

### DISCUSSION THREAD

**Discussion by gabsfredes** [community/community#44053] [GitHub Education] [answered]
### Select Topic Area

Bug

### Body

Hi!
I've seen that a lot of students (including myself) are having trouble with copilot for students. I have my student situation approved in 2021 and my copilot is not for free anymore.
For what i've been reading this is an error, but nobody shows the solution.

How do I fix it?

**Comment by Akash1134** [comment, marked as answer]
Hi there,

This week, we've had to take down some of our Copilot resources for maintenance. While we did our best to source extra capacity elsewhere, we've been forced to limit some free users for the period of time where one of our busiest models is out of commission.

We are also aware that some students are still experiencing issues with redeeming and accessing Copilot and that the team is currently working to resolve this. We'll update the community when we have more information on that incident in a separate discussion.

We expect that this will cease to impact your use of Copilot after the 16th of January. In the meantime, you will have less frequent suggestions, but you should still be getting suggestions - just with rate limits in between. You may see these errors specifically if you look in the output pane or logs for your IDE.

Thanks for your patience and apologies for the inconvenience caused.

**Reply by alex9947** [reply]
Hello I am a student and am still getting this error message.

![image](https://user-images.githubusercontent.com/44882419/214123202-a6e272f9-a8ee-4694-b884-9842c37c8bfb.png)

**Reply by iarata** [reply]
I am having the same issue now. Any updates regarding the situation?

**Reply by ajnart** [reply]
> I am having the same issue now. Any updates regarding the situation?

Mine just fixed itself the next day

**Comment by Branko4453** [comment]
Hello, 

I've been using Copilot for 1 year now and last week I got a message that my free Copilot has expired. I'm a student so I don't understand why my subscription is not free anymore. What should I do, to get free student subscription back?

Kind regards.

**Comment by ajnart** [comment]
Having the same issue as @Branko4453 above, I just got the message about my trial ending but all the documentation indicates that Copilot should stay free for me as long as I am a student

**Comment by Erenkrmt** [comment]
Having the same issue as all the other People above.

**Comment by datacubeR** [comment]
Same in here, no matter how install or uninstall vscode extension it just not suggest any code. And my config says my extended access has ended but I'm eligible to use Copilot for free because I have my student situation up to date, but can make the extension work...

**Comment by LameckMeshack** [comment]
I also have the same issue I can't access it completely yetI am a verified student

**Reply by davecheney** [reply]
@LameckMeshack i have verified that your 2024 coupon has been applied to your account and you have Copilot access.

**Reply by LameckMeshack** [reply]
@davecheney Yes it is now working perfectly

**Comment by nicobrch** [comment]
Same here, I have all student pack benefits, like github pro, but VSCode extensions shows that my account has no acces to Github Copilot. .😢

**Reply by davecheney** [reply]
@nicobrch your student coupon expired last month.

**Reply by nicobrch** [reply]
Thanks, I'll re apply now. We should get an e-mail for this tho

**Reply by davecheney** [reply]
I agree. Unfortunately I don't have access to that system to confirm if one was delivered.

**Comment by uwu-alexandra** [comment]
Same here, any solution for this?

**Reply by davecheney** [reply]
@uwu-alexandra your student coupon expired yesterday

**Reply by uwu-alexandra** [reply]
oh silly me, thanks and sorry

**Reply by davecheney** [reply]
np, if you're a current student then you can re-apply and you will get a new coupon which is good for two years.

**Comment by Sof2222** [comment]
Maybe a slightly different situation to the others here..
I am a student and in github itself it says I have copilot access but when I am in vscode it says 'get started with githubcopilot start 30-day free trial'.. I am unsure why that is?
I have tried uninstalling githubcopilot as well as toggling the 'manage trusted extentions' on it under the profile part of vs code + restarted a few times but nothing is fixing it..
Any suggestions?
(Note I was having a diferent error today where I was getting this error: "copilot Cannot read properties of undefined (reading 'split')" most of the time when trying to use copilot so updated it and lost access after I updated it).
<img width="661" alt="Screenshot 2024-02-12 at 09 44 44" src="https://github.com/community/community/assets/131628600/47e723af-a4da-4049-9936-6a0d3d97c182">
![Screenshot 2024-02-12 at 09 44 49](https://github.com/community/community/assets/131628600/dd73ed12-7bce-453e-be04-1017e01796d2)

**Reply by davecheney** [reply]
@Sof2222 please create a new discussion.

**Reply by CERNOJ** [reply]
@Sof2222 how did you manage to solve your issue? I am facing the same. thanks!

**Reply by Sof2222** [reply]
Not really solved but I have the account with copilot access logged into vs code and then logged into my other git account with the terminal:
Run: gh auth login

This means I can push with git from one account and use github from another. Havent found any way at all to meaningfully be logged into two accounts (I can be logged in but one will dominate all git and I cannot swap them without logging in and out which is tiresom)

**Reply by Sof2222** [reply]
The reason for it (my issue at least) is because vs code is dodgy with logging into 2 accounts at once

**Comment by driver08-str** [comment]
![Sin título](https://github.com/community/community/assets/159456755/ae022f02-2db3-4f6e-845d-2e533f0b0ff7)


Quiciera saber que esta sucediendo y que pasos debo seguir para que funcione de manera correcta. muchas gracias!

**Reply by davecheney** [reply]
@driver08-str please create a new discussion.

**Reply by driver08-str** [reply]
como lo hago?

**Comment by dyardy** [comment]
I have been approved for Copilot access via Student pack, however how do I enable for my account?

**Reply by makukhin-ie** [reply]
+1. Have the same issue. Copilot did not start and noticed that when tried to go through Primer: Copilot and Primer: Codespaces tasks. If for codespaces it is not required, for the other task I can just imitate it.

Previously had a Copilot subscription for a few months. Could that be the reason?

**Comment by Nurxat21** [comment]
I have same issue, any solutions here?

**Comment by Mertcyigit** [comment]
I have the same issue, any solutions?

**Comment by spriglithika** [comment]
I have been granted GitHub Student and received benefits, and can see that I have access to GitHub Pro, but when I go to Copilot under Code, planning, and automation, it only offers for me to start a free trial instead of saying I have access.

**Comment by yorushikanbuna** [comment]
![{BF7AE4A7-7E5B-4691-BF8F-5B0D7E9BB193}](https://github.com/community/community/assets/51018951/6e08016f-a507-4263-b953-7dcd1973b009)
 same issue,  I can't use copilot.

**Comment by qiyan98** [comment]
Hi there,

I re-verified my student status lately but copilot is still not working. Can you please take a look? @davecheney @Akash1134 

Many thanks!

**Reply by samus-aran** [reply]
Hi @qiyan98 

- This discussion is closed. Please do not revive the conversation as it is most likely not relevant anymore.
- Please note, contacting any member of GitHub Staff via unsolicited mentions or pings is strongly discouraged and may be considered a violation of our prohibition against harassment. Please review our [Code of Conduct](https://docs.github.com/en/site-policy/github-terms/github-community-code-of-conduct) for more information.
Let's work together to keep GitHub Community a place where people feel safe to participate by being respectful of them and their time.
Any repeat and future violations may result in a temporary or indefinite block from the Community. Thanks for understanding.
- Please open up a new discussion to address your query. Have you looked at the [Copilot FAQ](https://github.com/orgs/community/discussions/47318) or [Student Developer Pack FAQ?](https://github.com/orgs/community/discussions/111352)

**Comment by Nicat-dcw** [comment]
Hello. I have student pack since 2023 and then (today) i cant use copilot for free. What's issue ? Am i doing something wrong? Or something went wrong?
